v2client/algod/v2wrappers.go

```diff
@@ -0,0 +1,118 @@
+	} else if params.Format == "msgpack" {
+		err = client.get(ctx, &result, fmt.Sprintf("/transactions/pending/%s", txid), params, headers)
+	} else {
+		err = fmt.Errorf("unrecognized format %s, valid formats are json or msgpack", params.Format)
+	}
+	return
+}
+
+func (client Client) SendRawTransaction(ctx context.Context, txBytes []byte, headers ...*common.Header) (txid string, err error) {
+	var response models.TxId
+	err = client.post(ctx, &response, "/transactions", nil, headers)
+	txid = string(response)
+	return
+}
+
+func (client Client) PendingTransactionsByAddress(ctx context.Context, account string, params models.GetPendingTransactionsByAddressParams, headers ...*common.Header) (result []types.Transaction, err error) {
+	err = client.get(ctx, &result, fmt.Sprintf("/accounts/%s/transactions/pending", account), params, headers)
+	if params.Format == "json" {
+		var response []models.Transaction
+		err = client.get(ctx, &response, fmt.Sprintf("/accounts/%s/transactions/pending", account), params, headers)
+		// TODO built result from response
+	} else if params.Format == "msgpack" {
+		err = client.get(ctx, &result, fmt.Sprintf("/accounts/%s/transactions/pending", account), params, headers)
+	} else {
+		err = fmt.Errorf("unrecognized format %s, valid formats are json or msgpack", params.Format)
+	}
+	return
+}
+
+func (client Client) Status(ctx context.Context, headers ...*common.Header) (status models.NodeStatus, err error) {
+	err = client.get(ctx, &status, "/status", nil, headers)
+	return
+}
+
+func (client Client) Supply(ctx context.Context, headers ...*common.Header) (supply models.Supply, err error) {
+	err = client.get(ctx, &supply, "/ledger/supply", nil, headers)
+	return
+}
+
+func (client Client) StatusAfterBlock(ctx context.Context, round uint64, headers ...*common.Header) (status models.NodeStatus, err error) {
+	err = client.get(ctx, &status, fmt.Sprintf("/statuswait-for-block-after/%d", round), nil, headers)
+	return
+}
+
+func (client Client) AccountInformation(ctx context.Context, address string, headers ...*common.Header) (result models.Account, err error) {
+	err = client.get(ctx, &result, fmt.Sprintf("/accounts/%s", address), nil, headers)
+	return
+}
+
+func (client Client) Block(ctx context.Context, round uint64, params models.GetBlockParams, headers ...*common.Header) (result models.Block, err error) {
+	err = client.get(ctx, &result, fmt.Sprintf("/block/%d", round), params, headers)
+	if params.Format == "json" {
+		var response models.RawBlockJson
+		err = client.get(ctx, &response, fmt.Sprintf("/block/%d", round), params, headers)
+		// TODO built result from response
+	} else if params.Format == "msgpack" {
+		var response models.RawBlockMsgpack
+		err = client.get(ctx, &response, fmt.Sprintf("/block/%d", round), params, headers)
+		// TODO built result from response
+	} else {
+		err = fmt.Errorf("unrecognized format %s, valid formats are json or msgpack", params.Format)
+	}
+	return
+}
```

Comment:
@winder the schema only says `RawBlock` might come back, but it might come back in two different formats. Is this strategy of attempting to decode into two different structs appropriate?

Reply:
No, it depends on the user not the server. The users should specify what format they want it. See `format` query parameter in the spec.